commit 5650f16e71945cc2bda8b2602faf788e4571ec3d (refs/changes/31/5031/6)
Author: John Levon <john.levon@joyent.com>
Date:   2018-11-13T03:33:39+00:00 (11 months ago)
    
    OS-7336 lx_acl_getxattr leaks the ACL
    Reviewed by: Patrick Mooney <patrick.mooney@joyent.com>
    Reviewed by: Robert Mustacchi <rm@joyent.com>
    Approved by: Robert Mustacchi <rm@joyent.com>

diff --git a/usr/src/uts/common/brand/lx/os/lx_acl.c b/usr/src/uts/common/brand/lx/os/lx_acl.c
index 594671f401..184f05b6ed 100644
--- a/usr/src/uts/common/brand/lx/os/lx_acl.c
+++ b/usr/src/uts/common/brand/lx/os/lx_acl.c
@@ -10,7 +10,7 @@
  */
 
 /*
- * Copyright (c) 2017 Joyent, Inc.
+ * Copyright (c) 2018, Joyent, Inc.
  */
 
 #include <sys/types.h>
@@ -192,6 +192,9 @@ lx_acl_getxattr(vnode_t *vp, enum lx_acl_type atype, void *data, size_t slen,
 		return (err);
 	}
 
+	if (vsattr.vsa_aclentp != NULL)
+		kmem_free(vsattr.vsa_aclentp, vsattr.vsa_aclentsz);
+
 	return (ENODATA);
 }
 
